src/procedural-macros.md

```diff
@@ -318,45 +318,75 @@ r[macro.proc.token]
 ## Declarative macro tokens and procedural macro tokens
 
 r[macro.proc.token.intro]
-Declarative `macro_rules` macros and procedural macros use similar, but
-different definitions for tokens (or rather [`TokenTree`s].)
-
-r[macro.proc.token.macro_rules]
-Token trees in `macro_rules` (corresponding to `tt` matchers) are defined as
-- Delimited groups (`(...)`, `{...}`, etc)
-- All operators supported by the language, both single-character and
-  multi-character ones (`+`, `+=`).
-    - Note that this set doesn't include the single quote `'`.
-- Literals (`"string"`, `1`, etc)
-    - Note that negation (e.g. `-1`) is never a part of such literal tokens,
-      but a separate operator token.
-- Identifiers, including keywords (`ident`, `r#ident`, `fn`)
-- Lifetimes (`'ident`)
-- Metavariable substitutions in `macro_rules` (e.g. `$my_expr` in
-  `macro_rules! mac { ($my_expr: expr) => { $my_expr } }` after the `mac`'s
-  expansion, which will be considered a single token tree regardless of the
-  passed expression)
+Declarative `macro_rules` macros and procedural macros use similar but different definitions for tokens. Token trees in `macro_rules` (corresponding to `tt` matchers) are defined as the [TokenTree] production. Token trees for procedural macros are described below.
 
 r[macro.proc.token.tree]
-Token trees in procedural macros are defined as
-- Delimited groups (`(...)`, `{...}`, etc)
-- All punctuation characters used in operators supported by the language (`+`,
-  but not `+=`), and also the single quote `'` character (typically used in
-  lifetimes, see below for lifetime splitting and joining behavior)
-- Literals (`"string"`, `1`, etc)
-    - Negation (e.g. `-1`) is supported as a part of integer
-      and floating point literals.
-- Identifiers, including keywords (`ident`, `r#ident`, `fn`)
+Token trees in procedural macros are defined as:
+
+```grammar,macros
+@root ProcMacroTokenTree -> ProcMacroToken | ProcMacroDelimTokenTree
+
+ProcMacroDelimTokenTree ->
+      `(` ProcMacroTokenTree* `)`
+    | `[` ProcMacroTokenTree* `]`
+    | `{` ProcMacroTokenTree* `}`
+
+ProcMacroToken ->
+      ProcMacroTokenIdent
+    | ProcMacroTokenPunct
+    | ProcMacroTokenLiteral
+
+ProcMacroTokenIdent -> IDENTIFIER_OR_KEYWORD | RAW_IDENTIFIER | `_`
```

Comment:
Doesn't reference consider `_` a keyword?
I think it should, it's lexically an identifier, and it's reserved, so it's a keyword (== reserved identifier).